src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1133,7 +1133,22 @@ private uint getClassSize(CORINFO_CLASS_STRUCT_* cls)
         }
 
         private uint getClassAlignmentRequirement(CORINFO_CLASS_STRUCT_* cls, bool fDoubleAlignHint)
-        { throw new NotImplementedException("getClassAlignmentRequirement"); }
+        {
+            if (_compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.ARM ||
+                _compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.ARMEL)
+            {
+                // Default alignment is sizeof(void*)
+                uint result = (uint)sizeof(void*);
+                var type = HandleToObject(cls);
+                MetadataType typeDef = type.GetTypeDefinition() as MetadataType;
```

Comment:
The alignment lives on DefType, so you should be just casting to DefType here - without the rest. Also, the fallback to pointer-size alignment should not happen, and it is probably incorrect if it happens. It may better to just let it throw exception so that we noticed. Taking these two together makes the implementation:

`return ((DefType)type).InstanceByteAlignment.AsInt;`